ui/job-view/details/DetailsPanel.jsx

```diff
@@ -0,0 +1,521 @@
+  }
+
+  async updateClassifications(job) {
+    const classifications = await JobClassificationModel.getList({ job_id: job.id });
+    const bugs = await BugJobMapModel.getList({ job_id: job.id });
+    this.setState({ classifications, bugs });
+  }
+
+  selectJob(newJob) {
+    const { repoName } = this.props;
+
+    if (this.selectJobController !== null) {
+      // Cancel the in-progress fetch requests.
+      this.selectJobController.abort();
+    }
+    // eslint-disable-next-line no-undef
+    this.selectJobController = new AbortController();
+
+    let jobDetails = [];
+    const jobPromise = JobModel.get(
+      repoName, newJob.id,
+      this.selectJobController.signal);
+
+    const jobDetailPromise = JobDetailModel.getJobDetails(
+      { job_guid: newJob.job_guid },
+      this.selectJobController.signal);
+
+    const jobLogUrlPromise = JobLogUrlModel.getList(
+      { job_id: newJob.id },
+      this.selectJobController.signal);
+
+    const phSeriesPromise = this.PhSeries.getSeriesData(
+      repoName, { job_id: newJob.id });
+
+    Promise.all([
+      jobPromise,
+      jobDetailPromise,
+      jobLogUrlPromise,
+      phSeriesPromise
+    ]).then(async (results) => {
+
+      // The first result comes from the job promise.
+      // This version of the job has more information than what we get in the main job list.  This
+      // is what we'll pass to the rest of the details panel.  It has extra fields like
+      // taskcluster_metadata.
+      const job = results[0];
+      const jobRevision = this.ThResultSetStore.getPush(job.result_set_id).revision;
+
+      // the second result comes from the job detail promise
+      jobDetails = results[1];
+
+      // incorporate the buildername into the job details if this is a buildbot job
+      // (i.e. it has a buildbot request id)
+      const buildbotRequestIdDetail = jobDetails.find(detail => detail.title === 'buildbot_request_id');
+      if (buildbotRequestIdDetail) {
+        jobDetails = [...jobDetails, { title: 'Buildername', value: job.ref_data_name }];
+      }
+
+      // the third result comes from the jobLogUrl promise
+      // exclude the json log URLs
+      const jobLogUrls = results[2].filter(log => !log.name.endsWith('_json'));
+
+      let logParseStatus = 'unavailable';
+      // Provide a parse status as a scope variable for logviewer shortcut
+      if (jobLogUrls.length && jobLogUrls[0].parse_status) {
+        logParseStatus = jobLogUrls[0].parse_status;
+      }
+
+      // Provide a parse status for the model
+      const jobLogsAllParsed = (jobLogUrls ?
+        jobLogUrls.every(jlu => jlu.parse_status !== 'pending') :
+        false);
+
+      const lvUrl = getLogViewerUrl(job.id, repoName);
+      const lvFullUrl = location.origin + '/' + lvUrl;
+      const reftestUrl = jobLogUrls.length ?
+        `${getReftestUrl(jobLogUrls[0].url)}&only_show_unexpected=1` :
+        '';
+      const performanceData = flatten(Object.values(results[3]));
```

Comment:
After rebasing on `master` to resolve the conflict in `ui/plugins/controller.js ` (from #3627), could you convert this and the other `flatten()` usage to ES6? :-)

Reply:
Absolutely, yeah.  Done.